app.py

```diff
@@ -0,0 +1,437 @@
+    """
+    return json.dumps(
+        {
+            field_maps[environment]["messages_invoice_id"]: knack_invoice,
+            field_maps[environment]["messages_connected_invoice"]: knack_invoice,
+            field_maps[environment]["messages_created_date"]: today_date,
+            field_maps[environment]["messages_status"]: payment_status,
+            field_maps[environment]["messages_citybase_id"]: citybase_id,
+        }
+    )
+
+
+def update_parent_reservation(environment, knack_record_id, today_date):
+    """
+    Uses a knack_record_id to look up the transaction in knack, then finds the appropriate connection field
+    And sends payload to knack, marking parent reservation payment received status as TRUE
+    """
+    record_response = requests.get(
+        f"{KNACK_API_URL}{TRANSACTIONS_OBJECT_ID}/records/{knack_record_id}",
+        headers=headers[environment],
+    )
+    record_data = record_response.json()
+    if record_data[field_maps[environment]["ots_connection_field"]]:
+        # get parent record id
+        parent_record_response = record_data[
+            field_maps[environment]["ots_connection_field"] + "_raw"
+        ]
+        parent_record_id = parent_record_response[0]["id"]
+        ots_payload = json.dumps(
+            {
+                field_maps[environment]["ots_application_status"]: "Approved",
+                field_maps[environment]["ots_paid_status"]: True,
+                field_maps[environment]["ots_payment_date"]: today_date,
+            }
+        )
+        parent_update_response = requests.put(
+            f"{KNACK_API_URL}{OTS_OBJECT_ID}/records/{parent_record_id}",
+            headers=headers[environment],
+            data=ots_payload,
+        )
+        print(parent_update_response)
+    if record_data[field_maps[environment]["lpb_connection_field"]]:
+        # get parent record id
+        parent_record_response = record_data[
+            field_maps[environment]["lpb_connection_field"] + "_raw"
+        ]
+        parent_record_id = parent_record_response[0]["id"]
+        lpb_payload = json.dumps(
+            {
+                field_maps[environment]["lpb_application_status"]: "Approved",
+                field_maps[environment]["lpb_paid_status"]: True,
+                field_maps[environment]["lpb_payment_date"]: today_date,
+            }
+        )
+        parent_update_response = requests.put(
+            f"{KNACK_API_URL}{LPB_OBJECT_ID}/records/{parent_record_id}",
+            headers=headers[environment],
+            data=lpb_payload,
+        )
+        print(parent_update_response)
+
+
+@app.route("/")
+def index():
+    now = datetime.now().isoformat()
+    return f"Austin Transportation Public Works Department Citybase healthcheck {now}"
+
+
+@app.route("/citybase_postback", methods=["POST"])
+def handle_postback():
+    today_date = datetime.now().strftime("%m/%d/%Y %H:%M")
+    citybase_data = request.get_json()
```

Comment:
does this method throw if the request is not valid JSON? does there need to be a handler here to gracefully deal with that?

this also looks like a spot where we'd want to safely validate the JSON payload upfront. e.g looks like we're expecting a `"data"` property with a `total_amount` that is a float with two decimal places?

is there any authentication or key exchange that is preventing an anonymous client to modify records via this endpoint?